### ISSUE TITLE
Ubuntu 18.04 - Issues and curl: (7) Failed to connect to localhost port 8888: Connection refused

### ISSUE THREAD

**Issue by brianphobos** [EOSIO/eosio-project-demux-example#25]
There are several dependency issues that Ubuntu 18.04 gets hung up on when running the scripts. This could help someone build on Ubuntu faster.   

**You need the following:**  nodemon, dotenv, node-sass, node-sass-chokidar, lodash.assign, true-case-path

sudo npm install -g nodemon
sudo npm install -g dotenv
sudo npm install -g node-sass (Probably won't work.......I had to copy those libraries from a folder I had from an Ionic install)
sudo npm i --save lodash.assign
sudo npm i --save true-case-path

Resolving all that got me all the way trough to where everything was running including the front end.   
![frontend-running](https://user-images.githubusercontent.com/40128659/44989875-2b116f80-af44-11e8-99d6-3042a8efda51.png)
**The issue thought is that it won't write to the blocks because for some reason it won't connect to port 8888.**   
![failed-to-connect-8888](https://user-images.githubusercontent.com/40128659/44989960-7297fb80-af44-11e8-89b7-8659ef17a57b.png)
### Here are the results of 'sudo netstat -tulpn' when it isn't producing blocks.   
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      992/systemd-resolve 
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      9228/sshd           
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      1148/cupsd          
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN      4620/node           
tcp        0      0 10.0.0.9:10010          0.0.0.0:*               LISTEN      14334/docker-contai 
tcp6       0      0 :::22                   :::*                    LISTEN      9228/sshd           
tcp6       0      0 ::1:631                 :::*                    LISTEN      1148/cupsd          
udp        0      0 0.0.0.0:35677           0.0.0.0:*                           1140/avahi-daemon:  
udp     4608      0 127.0.0.53:53           0.0.0.0:*                           992/systemd-resolve 
udp        0      0 0.0.0.0:68              0.0.0.0:*                           1676/dhclient       
udp        0      0 0.0.0.0:631             0.0.0.0:*                           1273/cups-browsed   
udp     6912      0 0.0.0.0:5353            0.0.0.0:*                           3166/chrome         
udp    12288      0 0.0.0.0:5353            0.0.0.0:*                           1140/avahi-daemon:  
udp6       0      0 :::44243                :::*                                1140/avahi-daemon:  
udp6   11520      0 :::5353                 :::*                                3166/chrome         
udp6   50176      0 :::5353                 :::*                                1140/avahi-daemon:

## Here are the results of 'sudo netstat -tulpn' when it is producing blocks.  

Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      992/systemd-resolve 
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      9228/sshd           
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      1148/cupsd          
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN      4620/node           
tcp        0      0 10.0.0.9:10010          0.0.0.0:*               LISTEN      14334/docker-contai 
tcp6       0      0 :::9876                 :::*                    LISTEN      14757/docker-proxy  
tcp6       0      0 :::22                   :::*                    LISTEN      9228/sshd           
tcp6       0      0 ::1:631                 :::*                    LISTEN      1148/cupsd        
tcp6       0      0 :::8888                 :::*                    LISTEN      14770/docker-proxy  
udp        0      0 0.0.0.0:35677           0.0.0.0:*                           1140/avahi-daemon:  
udp    45312      0 127.0.0.53:53           0.0.0.0:*                           992/systemd-resolve 
udp        0      0 0.0.0.0:68              0.0.0.0:*                           1676/dhclient       
udp        0      0 0.0.0.0:631             0.0.0.0:*                           1273/cups-browsed   
udp     9216      0 0.0.0.0:5353            0.0.0.0:*                           3166/chrome         
udp     9216      0 0.0.0.0:5353            0.0.0.0:*                           3166/chrome         
udp    21504      0 0.0.0.0:5353            0.0.0.0:*                           1140/avahi-daemon:  
udp6       0      0 :::44243                :::*                                1140/avahi-daemon:  
udp6   19968      0 :::5353                 :::*                                3166/chrome         
udp6   19968      0 :::5353                 :::*                                3166/chrome         
udp6   19968      0 :::5353                 :::*                                3166/chrome         
udp6   26624      0 :::5353                 :::*                                1140/avahi-daemon:  

#### curl -v 127.0.0.1:8888
* Rebuilt URL to: 127.0.0.1:8888/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8888 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8888
> User-Agent: curl/7.58.0
> Accept: */*
> 
* Recv failure: Connection reset by peer
* stopped the pause stream!
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer

### I'm sort of stumped at this point.  Thanks in advance for looking at this.

**Comment by cleett** [comment]
Use eosio/eos-dev:v1.2.3 instead. It seems solve the problem.

**Comment by brianphobos** [comment]
### Update:  Another user figured out to use eosio/eos-dev:v1.2.3 instead

If you are running Ubuntu 18.04 you can update line 24 of first_time_setup.sh to pull v1.2.3 instead of 1.2.2 and line 20 of start_eosio_docker.sh to v1.2.3 as well.   

After doing that some of the dependencies might have to be resolved as I mentioned above and then the only other weird thing is if node-sass won't install.   I had to copy that folder from an Ionic app folder and each time I tried to resolve a dependency it would get deleted again from the /frontend/node_modules.   So you could install Ionic and keep copying it in there until you get through all the errors.   Weird workaround but that is how I got through it.   

Good luck. Thanks

**Comment by TaraTritt** [comment]
Awesome. Thanks @brianphobos for the detailed setup instructions for Ubuntu 18.04 and verifying the resolution. I will open a PR soon to update the eosio/eos-dev to v1.2.3 to permanently resolve the port issue you were experiencing. As for the node-sass install issue, which version of node and npm are you running?

**Comment by brianphobos** [comment]
Thanks Tara!  
Node Version: v10.7.0
NPM Version: 6.2.0

**Comment by brianphobos** [comment]
Yeah I already have build-essential installed.   It is a really weird issue but other people who were working with other projects were running into the same thing with node-sass with Ubuntu 18.04.  So I'm not sure what the exact issue is.   The strangest thing is that it was working fine with my Ionic install.
